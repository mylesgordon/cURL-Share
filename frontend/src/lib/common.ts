export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}