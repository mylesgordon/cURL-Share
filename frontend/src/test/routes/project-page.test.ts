import ProjectPage from '../../routes/project/[slug]/+page.svelte';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';
import type { ComponentProps } from 'svelte';

describe('Project page', () => {
	type ProjectPageProps = ComponentProps<ProjectPage>;
	const labels = ['IPB-xxxx', 'Feature', 'User Journey'];
	const commonProps = { labels, projectId: '1', projectName: 'Cool project' };

	const propsWithEntries: ProjectPageProps = {
		data: {
			...commonProps,
			groups: [
				{
					groupId: 1,
					title: 'Login Journey',
					description: 'cURL group defining a user journey!',
					labels: [labels[1], labels[2]]
				},
				{
					groupId: 2,
					title: 'IPB-1293',
					description:
						'Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text ',
					labels: [labels[0], labels[1], labels[2]]
				}
			]
		}
	};

	const propsWithNoEntries: ProjectPageProps = { data: { ...commonProps, groups: [] } };

	testMatchingSnapshot(ProjectPage, propsWithEntries);
	testMatchingSnapshot(ProjectPage, propsWithNoEntries);

	it('should display the correct title', () => {
		render(ProjectPage, propsWithEntries);
		expect(document.title).toEqual('Cool project | cURL Share');
	});
});
