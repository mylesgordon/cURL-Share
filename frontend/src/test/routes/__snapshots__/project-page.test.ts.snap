// Vitest Snapshot v1

exports[`Project page > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-row gap-2 justify-between items-center"
    >
      <div
        class="flex gap-2 items-center"
      >
        <h1>
          Cool project
        </h1>
         
        <a
          aria-label="Create project"
          href="/project/1/group/create"
        >
          <svg
            height="1.2em"
            viewBox="0 0 1408 1408"
            width="1.2em"
          >
            <path
              d="M1408 608v192q0 40-28 68t-68 28H896v416q0 40-28 68t-68 28H608q-40 0-68-28t-28-68V896H96q-40 0-68-28T0 800V608q0-40 28-68t68-28h416V96q0-40 28-68t68-28h192q40 0 68 28t28 68v416h416q40 0 68 28t28 68z"
              fill="currentColor"
            />
          </svg>
          <!--&lt;Plus&gt;-->
        </a>
        <!--&lt;IconLink&gt;-->
         
        <a
          aria-label="Project settings"
          href="/project/1/settings"
        >
          <svg
            height="1.2em"
            viewBox="0 0 1536 1536"
            width="1.2em"
          >
            <path
              d="M1024 768q0-106-75-181t-181-75t-181 75t-75 181t75 181t181 75t181-75t75-181zm512-109v222q0 12-8 23t-20 13l-185 28q-19 54-39 91q35 50 107 138q10 12 10 25t-9 23q-27 37-99 108t-94 71q-12 0-26-9l-138-108q-44 23-91 38q-16 136-29 186q-7 28-36 28H657q-14 0-24.5-8.5T621 1506l-28-184q-49-16-90-37l-141 107q-10 9-25 9q-14 0-25-11q-126-114-165-168q-7-10-7-23q0-12 8-23q15-21 51-66.5t54-70.5q-27-50-41-99L29 913q-13-2-21-12.5T0 877V655q0-12 8-23t19-13l186-28q14-46 39-92q-40-57-107-138q-10-12-10-24q0-10 9-23q26-36 98.5-107.5T337 135q13 0 26 10l138 107q44-23 91-38q16-136 29-186q7-28 36-28h222q14 0 24.5 8.5T915 30l28 184q49 16 90 37l142-107q9-9 24-9q13 0 25 10q129 119 165 170q7 8 7 22q0 12-8 23q-15 21-51 66.5t-54 70.5q26 50 41 98l183 28q13 2 21 12.5t8 23.5z"
              fill="currentColor"
            />
          </svg>
          <!--&lt;Gear&gt;-->
        </a>
        <!--&lt;IconLink&gt;-->
      </div>
       
      <div
        class="flex s-MMN3rRE-BmEv"
      >
        <div
          class="w-100 s-JZthjkXlZrq3"
        >
          <label
            class="label screenreader-only s-JZthjkXlZrq3"
            for=""
          >
            
          </label>
           
          <input
            class="input s-JZthjkXlZrq3"
            id=""
            placeholder="Search..."
            type="text"
          />
           
        </div>
        <!--&lt;Input&gt;-->
         
        <select
          class="s-MMN3rRE-BmEv"
          id="labels"
          name="labels"
        >
          <option
            class="s-MMN3rRE-BmEv"
            value=""
          >
            Choose a label
          </option>
          <option
            class="s-MMN3rRE-BmEv"
            value="Feature"
          >
            Feature
          </option>
          <option
            class="s-MMN3rRE-BmEv"
            value="User Journey"
          >
            User Journey
          </option>
          <option
            class="s-MMN3rRE-BmEv"
            value="IPB-xxxx"
          >
            IPB-xxxx
          </option>
        </select>
      </div>
      <!--&lt;Search&gt;-->
    </div>
     
    <!--&lt;Meta&gt;-->
     
    
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 flex-grow full truncate text-center project-title s-kSbj4w364X09"
        data-testid="curl-group-list-entry-1-title"
        href="/project/1/group/1"
      >
        Login Journey
      </a>
       
      <p
        class="p-4 flex-grow project-description s-kSbj4w364X09"
        data-testid="curl-group-list-entry-1-description"
      >
        cURL group defining a user journey!
      </p>
       
      <div
        aria-label="Project labels"
        class="flex flex-grow flex-wrap justify-center project-label-container s-kSbj4w364X09"
      >
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-1-label-Feature"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            Feature
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-1-label-User Journey"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            User Journey
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
      </div>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;CurlGroupListEntry&gt;-->
    
    <div
      class="card card-shadow card-rounded card-border mt-4 s-CTyDWs24qi3Z"
    >
      <a
        class="p-4 flex-grow full truncate text-center project-title s-kSbj4w364X09"
        data-testid="curl-group-list-entry-2-title"
        href="/project/1/group/2"
      >
        IPB-1293
      </a>
       
      <p
        class="p-4 flex-grow project-description s-kSbj4w364X09"
        data-testid="curl-group-list-entry-2-description"
      >
        Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text Lorem ipsum this is long text 
      </p>
       
      <div
        aria-label="Project labels"
        class="flex flex-grow flex-wrap justify-center project-label-container s-kSbj4w364X09"
      >
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-2-label-IPB-xxxx"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            IPB-xxxx
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-2-label-Feature"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            Feature
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
        <span
          class="s-kSbj4w364X09"
          data-testid="curl-group-list-entry-2-label-User Journey"
        >
          <span
            class="tag tag-pill s-1BpX-UQ-3FT-"
          >
            User Journey
          </span>
          <!--&lt;Tag&gt;-->
           
        </span>
      </div>
    </div>
    <!--&lt;Card&gt;-->
    <!--&lt;CurlGroupListEntry&gt;-->
    
    
    
    <!--&lt;+page&gt;-->
  </div>
</body>
`;

exports[`Project page > should match snapshot 2`] = `
<body>
  <div>
    <div
      class="flex flex-row gap-2 justify-between items-center"
    >
      <div
        class="flex gap-2 items-center"
      >
        <h1>
          Cool project
        </h1>
         
      </div>
       
      <div
        class="flex s-MMN3rRE-BmEv"
      >
        <div
          class="w-100 s-JZthjkXlZrq3"
        >
          <label
            class="label screenreader-only s-JZthjkXlZrq3"
            for=""
          >
            
          </label>
           
          <input
            class="input s-JZthjkXlZrq3"
            id=""
            placeholder="Search..."
            type="text"
          />
           
        </div>
        <!--&lt;Input&gt;-->
         
      </div>
      <!--&lt;Search&gt;-->
    </div>
     
    <!--&lt;Meta&gt;-->
     
    
    
    
    <!--&lt;+page&gt;-->
  </div>
</body>
`;
